README.md

```diff
@@ -25,34 +25,113 @@ Install `make` on your system.
 
 Decide which image you want to build and run:
 
-`make -C images <local|mainnet|testnet>`
+`make -C images <mainnet|testnet>`
 
 E.g. to build the testnet image:
 
 ```
 make -C images testnet
 ```
 
-### Run the images
+### Run images using docker-compose
 
-`make <local|mainnet|testnet>`
+There are examples for testnet and mainnet in the `examples/`directory.
 
-E.g. to run a testnet image:
+#### Examples for testnet
+
+`cd examples/testnet/`
+
+By default the examples use the `latest` version. Make sure to pull it with:
 
 ```
-make testnet
+docker-compose pull
 ```
 
-#### pgAdmin
+Alternatively you can change the example `docker-compose.yml` file to use a specific version by changing the `image: lisk/testnet` line to `image: lisk/tesnet:<VERSION>`.
+For instance if you want to use version `0.9.12a` your `docker-compose.yml` should look like this:
 
-If you wish you can start pgAdmin by running `make pgadmin` in one of the subdirectories, e.g.:
+```
+version: "2"
+services:
 
+  lisk:
+      image: lisk/testnet:0.9.12a
+[...]
 ```
-cd testnet
-make pgadmin
+
+##### Start Lisk
+
+```
+make up
 ```
 
-***
+##### Rebuild database from blockchain snapshot
+
+```
+make coldstart
+```
+
+##### Rebuild database from latest blockchain snapshot
+
+Delete any previously downloaded blockchain.db.gz ensure the latest is downloaded:
+
+```
+make clean
+make coldstart
+```
+
+##### Stop Lisk
+
+```
+docker-compose stop
+```
+
+
+##### Delete all containers
+
+```
+docker-compose down --volumes
+```
+
+#### Upgrade
+
+Create a new container without exposing it, wait until it catches up, stop the old container and expose the new one.
+
+```
+cp -r examples/testnet/ new_testnet/
+cd new_testnet/
+# edit `docker-compose.yml` to not expose port 7000
+make coldstart
+# make sure everything works
+```
+
+#### Customize config.json
+
+Those environment variables that can be used to customize `config.json` inside a container:
+- `LISK_CONFIG_CONSOLE_LOG_LEVEL`
+- `LISK_CONFIG_FORGING_WHITELIST_IP`
+- `LISK_CONFIG_DB_HOST`
+- `LISK_CONFIG_DB_PORT`
+- `LISK_CONFIG_DB_DATABASE`
+- `LISK_CONFIG_DB_USER`
+- `LISK_CONFIG_DATABASE_PASSWORD`
+
+To customize other parts of `config.json` create your own image and add your custom config.json like so:
+
```

Comment:
It looks too complicated. I would remove these instructions to avoid support.

Reply:
It does? Note that this is optional and only necessary when users want to customize the config file themselves. I think it'll prevent support requests for adding configuration variables for less common parameters (see your own request above)